lib/elixir/lib/module.ex

```diff
@@ -861,6 +861,46 @@ defmodule Module do
     end, tuples)
   end
 
+  @spec make_overridable(module, module) :: :ok
+  def make_overridable(module, behaviour) when is_atom(module) and is_atom(behaviour) do
+    behaviour_definitions = :ets.lookup_element(data_table_for(module), :behaviour, 2)
+
+    error_explanation = cond do
+      !Code.ensure_compiled?(behaviour)                  -> "it was not defined"
+      !function_exported?(behaviour, :behaviour_info, 1) -> "it does not define any callbacks"
+      !:lists.member(behaviour, behaviour_definitions)   -> "its corresponding @behaviour #{inspect(behaviour)} was not specified"
```

Comment:
If we can easily check 1. then that's my vote. It's nice that I can look at the top of a module and see that it's a behaviour.